src/typescript/frontend/src/components/StatsBar.tsx

```diff
@@ -1,51 +1,324 @@
+      await timeout(1000);
+      const {
+        lastPrice,
+        change24h,
+        high24h,
+        low24h,
+        pairData,
+        lastPriceChange,
+        change24hPercent,
+      } = STATS_BAR_MOCK_DATA;
+      // END MOCK API CALL
+      return {
+        lastPrice: formatDecimal(lastPrice),
+        lastPriceChange: formatDecimalWithPlusMinus(lastPriceChange, 4),
+        change24h: formatDecimalWithPlusMinus(change24h, 4),
+        change24hPercent: formatDecimalWithPlusMinus(change24hPercent),
+        high24h: formatDecimal(high24h, 4),
+        low24h: formatDecimal(low24h, 4),
+        pairData: {
+          baseAsset: baseAsset,
+          quoteAsset: quoteAsset,
+          baseAssetIcon: pairData.baseAssetIcon,
+          quoteAssetIcon: pairData.quoteAssetIcon,
+          baseVolume: formatDecimal(pairData.baseVolume),
+          quoteVolume: formatDecimal(pairData.quoteVolume),
+        },
+      } as MarketStats;
+    },
+    { keepPreviousData: true, refetchOnWindowFocus: false }
+  );
+};
+
+const STATS_BAR_MOCK_DATA: MarketStats = {
+  lastPrice: "10.17",
+  lastPriceChange: "10.1738",
+  change24h: "-.9305",
+  change24hPercent: "-8.38",
+  high24h: "11.1681",
+  low24h: "9.85",
+  pairData: {
+    // asset names here don't matter for testing purposes
+    baseAsset: "doesn't",
+    quoteAsset: "matter",
+    baseAssetIcon: "/tokenImages/APT.png",
+    quoteAssetIcon: "/tokenImages/USD.png",
+    baseVolume: "6531688.77",
+    quoteVolume: "68026950.84",
+  },
+};
+
+// UTIL FUNCTIONS
+
+// format number to dollar
+const formatDecimal = (num: string, digits = 2): string => {
+  const roundedNum = Number(num).toFixed(digits);
+  const parts = roundedNum.split(".");
+  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+  return parts.join(".");
+};
+
+// formatDecimal but with + or - sign
+const formatDecimalWithPlusMinus = (num: string, digits = 2): string => {
+  const formattedNum = formatDecimal(num, digits);
+  return Number(num) >= 0 ? `+${formattedNum}` : formattedNum;
+};
+
+// color based on number, green if positive, red if negative
+const colorBasedOnNumber = (num: string | undefined): string => {
+  if (!num) return "text-green-500";
+
+  if (num[0] === "+") return "text-green-500";
+  if (num[0] === "-") return "text-red-500";
+
+  return Number(num) < 0 ? "text-red-500" : "text-green-500";
+};
+
+// COMPONENTS
+// leaving here until i learn more about how this project is structured
+type MarketIconPairProps = {
+  baseAssetIcon?: string;
+  quoteAssetIcon?: string;
+};
+const MarketIconPair = ({
+  baseAssetIcon = DEFAULT_TOKEN_ICON,
+  quoteAssetIcon = DEFAULT_TOKEN_ICON,
+}: MarketIconPairProps) => {
+  return (
+    <div className="relative flex">
+      {/* height width props required */}
+      <Image
+        src={baseAssetIcon}
+        alt="market-icon-pair"
+        width={40}
+        height={40}
+        className="z-[2] aspect-square  w-[30px] min-w-[30px] md:min-w-[40px]"
```

Comment:
I don't think you need the bracket syntax for the z indexes, and you can just use the classes provided by Tailwind. In general, you don't need to use brackets unless you need to assign a custom value. Here, `z-10` and `z-20` are fine.

https://tailwindcss.com/docs/z-index